_posts/2020-10-19-micrometer-metrics.adoc

```diff
@@ -35,36 +34,45 @@ The design and development of Micrometer focused on the needs of SREs from the v
 A key aspect to that approach is separating out the metric implementation into the registries,
 enabling different registries to customize how the metric export,
 aligning with a specific monitoring tool.
-Micrometer offers registries for Prometheus, DataDog, New Relic, Stackdriver, and many more.
-This enables the same application to switch the monitoring tool with a single dependency,
-and without adjusting application code utilizing metrics.
 
-Micrometer aligns itself with more modern monitoring solutions that utilize dimensional metrics,
-instead of hierarchical ones.
-Taking the example from the previous section would mean the metrics are:
+Micrometer was built to adapt to the changing cloud native landscape while presenting a steady API for applications and libraries to use.
+Each MeterRegistry implementation encapsulates its specific aggregation and reporting requirements.
+There are MeterRegistry implementations available for cloud monitoring systems like Prometheus and Statsd,
+as well as service providers like Datadog, New Relic or Stackdriver.
+You can include more than one of these registries at the same time, too.
 
-[source,shell script]
-----
-http_request_total{service="my-service", outcome="success"}
-http_request_total{service="my-service", outcome="failure"}
-----
+While Micrometer has an API for applications to use,
+practical usage is more focused on the MeterBinder abstraction.
+MeterBinders allow instrumentation for application libraries to be added later.
+It is even better when your application framework can register discovered MeterBinders on your behalf.
 
-We now have a single metric name,
-but with a series of labels or tags that identify the particular metric dimensions.
-Taking this approach makes slicing and dicing the data easier in monitoring systems.
-For instance, querying on `http_request_total` will provide a total count of all invocations.
-The labels, or dimensions,
-of `service` and `outcome` can be utilized to slice the data in different ways to provide insights for individual services
-or across services.
+Micrometer does follow different naming conventions than Dropwizard or MP Metrics, for a few reasons.
+First, it uses a dot notation for all metric names, e.g. my.metric.name.
+Each registry implementation can then apply its own naming convention to convert that name into something that matches its convention.
+The Prometheus MeterRegistry converts my.metric.name into my_metric_name for example.
```

Comment:
woops. This one needs a comma. ;)

The Prometheus MeterRegistry converts my.metric.name into my_metric_name, for example.